### ISSUE TITLE
[Issue] Message search requires exact keyword to produce results.

### ISSUE THREAD

**Issue by 8227846265** [vector-im/element-web#24697]
### Steps to reproduce

1. Go to a room
2. Click `Search` icon
3. Click `All Rooms`
4. Search a message, example `follower`
![image](https://user-images.githubusercontent.com/66582769/222036975-34d34f85-aa12-4c26-ae36-b1956234fc95.png)
5. Then add a single letter to the search query, example `followers`
![image](https://user-images.githubusercontent.com/66582769/222037159-3a6db439-9b3d-441c-a0d9-334a7e61bc6d.png)

#### What did you expect?
By searching `follower`, it should also show the search result for `followers`

#### What happened instead?
By searching `follower`, there's no search result.

In summary, it would make message search more efficient and easier to use if it doesn't follow an exact keyword search result logic, as the user doesn't have to try different variation of a single word per query just to get a search result.

### Operating system

Windows 11

### Application version

Element version: 1.11.24 Olm version: 3.2.12

### How did you install the app?

https://element.io/download

### Will you send logs?

No

**Comment by florianduros** [comment]
Closing in favour of https://github.com/vector-im/element-web/issues/24551
